File: JavaScript/leetcode_485.js
/* 
Given a binary array nums, return the maximum number of consecutive 1's in the array.
*/

var findMaxConsecutiveOnes = function(nums) {
    let maxN = 0;
    let temp = 0;
    for (let num of nums) {
        if (num ==1 ) {
            temp++;
            maxN = Math.max(maxN, temp);
        }
        else temp = 0
        
    };
    return maxN
};

File: Python/leetcode_485.py
'''
Given a binary array nums, return the maximum number
of consecutive 1's in the array.
'''


from typing import List


def find_max_consecutive_ones(nums: List[int]) -> int:
    max_number = 0
    temp_number = 0
    for num in nums:
        if num == 1:
            temp_number += 1
            max_number = max(max_number, temp_number)
        else:
            temp_number = 0
    return max_number
